app/models/chat.rb

```diff
@@ -6,4 +6,16 @@ class Chat < ApplicationRecord
   def users
     [user_a, user_b]
   end
+
+  def unread_messages_count(user)
+    messages.where(seen: false).where.not(user: user).count
+  end
+
+  def clean_unread_messages(user)
+    messages.where(seen: false).where.not(user: user).update_all(seen: true)
```

Comment:
The `messages.where(seen: false).where.not(user: user)` is repeated so it could be on a **scope** or in a **query object**